### ISSUE TITLE
[Discussion] Compile-time packages should be in devDependencies

### ISSUE THREAD

**Issue by jmarinaro** [jfurrow/flood#762]
## Discussion
The current docker build is bloated with unnecessary packages such as babel, webpack, and node-sass.
babel compiles js
webpack bundles static assets
node-sass compiles css

I propose we move these packages to devDependencies to slim down the final docker image.

**Comment by noraj** [comment]
I'm not sure I understand, but those are not dev dependencies like linter that are unneeded at runtime.
In fact babel, webpack, and node-sass are more some build tools, for example in ArchLinux pkgbuild this is called makedependencies, because they are not required for runtime but are required for building/making the soft/asset even in production or at each upgrade, not only for development.

**Comment by jmarinaro** [comment]
When I say runtime I am talking about the step of `npm start`, and none of the packages are necessary at that time. npm does not have a separate set of dependencies dedicated to build tools and the current convention is to place them into the devDependencies, which is evidenced by the docs of these modules install instructions containing `npm install --save-dev`

The change does not break the steps located in the README. The only place the `--production` flag is located in the project is in the Dockerfile. If you wanted to slim down your install outside of docker you could run:
```
npm install
npm run build
npm prune --production
npm start
```

**Comment by noraj** [comment]
Ok we'll wait @jfurrow to merge this.

**Comment by jfurrow** [comment]
It's difficult to know if dependencies should be in `dependencies` or `devDependencies`, but I agree with [this comment](https://github.com/webpack/webpack/issues/520#issuecomment-331689542) at the moment.

**Comment by noraj** [comment]
> tldr, Flood doesn't ship with static assets and these modules are required to build the static assets, so they belong in `dependencies`.

That is what I thought and told to @jmarinaro
